include/deal.II/distributed/shared_tria.h

```diff
@@ -0,0 +1,211 @@
+      typedef typename dealii::Triangulation<dim,spacedim>::active_cell_iterator active_cell_iterator;
+      typedef typename dealii::Triangulation<dim,spacedim>::cell_iterator        cell_iterator;
+
+      /**
+       * Constructor.
+       */
+      Triangulation (MPI_Comm mpi_communicator,
+                     const typename dealii::Triangulation<dim,spacedim>::MeshSmoothing =
+                       (dealii::Triangulation<dim,spacedim>::none) );
+
+      /**
+       * Destructor.
+       */
+      virtual ~Triangulation ();
+
+      /**
+       * Return locally owned subdomain id,
+       * which is equivalent to the rank of the current mpi process in
+       * communicator provided to class constructor.
+       */
+      virtual types::subdomain_id locally_owned_subdomain () const;
+
+      /**
+       * Return MPI communicator used by this triangulation.
+       */
+      virtual MPI_Comm get_communicator () const;
+
+      /**
+       * Coarsen and refine the mesh according to refinement and
+       * coarsening flags set.
+       *
+       * This step is equivalent to the dealii::Triangulation class
+       * with an addition of calling dealii::GridTools::partition_triangulation() at the end.
+       */
+      virtual void execute_coarsening_and_refinement ();
+
+      /**
+        * Create a triangulation.
+        *
+        * This function also partitions triangulation based on the
+        * MPI communicator provided to constructor.
+        */
+      virtual void create_triangulation (const std::vector< Point< spacedim > > &vertices,
+                                         const std::vector< CellData< dim > > &cells,
+                                         const SubCellData &subcelldata);
+
+      /**
+       * Return the number of active cells owned by each of the MPI
+       * processes that contribute to this triangulation. The element
+       * of this vector indexed by locally_owned_subdomain() equals
+       * the result of n_locally_owned_active_cells().
+       */
+      virtual const std::vector<unsigned int> &
+      n_locally_owned_active_cells_per_processor () const;
+
+      /**
+       * Return the number of active cells in the triangulation that
+       * are locally owned, i.e. that have a subdomain_id equal to
+       * locally_owned_subdomain().
+       *
+       */
+      virtual unsigned int n_locally_owned_active_cells () const;
+
+      /**
+       * Return the sum over all processors of the number of active
+       * cells owned by each processor. This equals the overall number
+       * of active cells in the shared triangulation.
+       */
+      virtual types::global_dof_index n_global_active_cells () const;
+
+      /**
+       * Returns the global maximum level. This may be bigger than n_levels.
+       */
+      virtual unsigned int n_global_levels () const;
+
+    private:
+      MPI_Comm mpi_communicator;
+      types::subdomain_id my_subdomain;
+      types::subdomain_id n_subdomains;
```

Comment:
Couldn't you push these (and the other member variables immediately below) into the base class `parallel::Triangulation`? It seems to me that these are the same for all parallel triangulations. This would also allow moving the member functions above that return these variables into the base class.